stacks-signer/src/v0/signer.rs

```diff
@@ -916,6 +952,14 @@ impl Signer {
 
         let mut block_info = match self.signer_db.block_lookup(block_hash) {
             Ok(Some(block_info)) => {
+                if block_info.reward_cycle != self.reward_cycle {
```

Comment:
Seems this check (here and elsewhere) could be folded into a variant of `self.sgner_db.block_lookup(block_hash)` that only returned `Some(block_info)` if the given reward cycle matched `block_info.reward_cycle`.  That would DRY things up.